modal/_container_entrypoint.py

```diff
@@ -587,32 +590,47 @@ async def call_function_async(
     try:
 
         async def run_input(input_id: str, function_call_id: str, args: Any, kwargs: Any) -> None:
-            output_index = SequenceNumber(0)  # mutable number we can increase from the generator loop
             started_at = time.time()
             reset_context = _set_current_context_ids(input_id, function_call_id)
-            async with function_io_manager.handle_input_exception.aio(input_id, started_at, output_index):
+            async with function_io_manager.handle_input_exception.aio(input_id, started_at):
                 res = imp_fun.fun(*args, **kwargs)
 
                 # TODO(erikbern): any exception below shouldn't be considered a user exception
                 if imp_fun.is_generator:
                     if not inspect.isasyncgen(res):
                         raise InvalidError(f"Async generator function returned value of type {type(res)}")
-                    async for value in res:
-                        await function_io_manager.push_generator_value.aio(
-                            input_id, started_at, output_index.value, value, imp_fun.data_format
+
+                    # Send up to this many outputs at a time.
+                    generator_queue: asyncio.Queue[Any] = await function_io_manager._queue_create.aio(1024)
+                    generator_output_task = asyncio.create_task(
```

Comment:
Should we use a `TaskContext` here instead?

Reply:
Not sure! The sync code path uses a synchronicity call, so it lives on the other event loop. This one uses an async synchronicity call. We might be able to remove the asyncio.create_task() here entirely actually, except due to a limitation of synchronicity where the `asyncio.run_coroutine_threadsafe()` call is placed inside another async function, so it actually doesn't make progress until awaited.

<img width="728" alt="image" src="https://github.com/modal-labs/modal-client/assets/7550632/4a3ad571-9810-43a9-a611-7e8c5a56b6a0">